src/agent/src/register.cpp

```diff
@@ -0,0 +1,142 @@
+            {
+                req.set(http::field::authorization, "Basic " + user_pass);
+            }
+
+            if (!body.empty())
+            {
+                req.set(http::field::content_type, "application/json");
+                req.body() = body;
+                req.prepare_payload();
+            }
+
+            http::write(socket, req);
+
+            beast::flat_buffer buffer;
+
+            http::read(socket, buffer, res);
+
+            std::cout << "Response code: " << res.result_int() << std::endl;
+            std::cout << "Response body: " << beast::buffers_to_string(res.body().data()) << std::endl;
+        }
+        catch (std::exception const& e)
+        {
+            std::cerr << "Error: " << e.what() << std::endl;
+            res.result(http::status::internal_server_error);
+            beast::ostream(res.body()) << "Internal server error: " << e.what();
+            res.prepare_payload();
+        }
+        return res;
+    }
+
+    std::pair<int, std::string>
+    SendAuthenticationRequest(const std::string& managerIp, const std::string& port, const std::string& user_pass)
+    {
+        http::response<http::dynamic_body> res =
+            sendRequest(managerIp, port, http::verb::post, "/authenticate", "", "", user_pass);
+        const auto token = beast::buffers_to_string(res.body().data());
+
+        return {res.result_int(), token};
+    }
+
+    int SendRegistrationRequest(const std::string& managerIp,
+                                const std::string& port,
+                                const std::string& token,
+                                const std::string& uuid,
+                                const std::optional<std::string>& name,
+                                const std::optional<std::string>& ip)
+    {
+        json bodyJson = {{"uuid", uuid}};
+
+        if (name.has_value())
+        {
+            bodyJson["name"] = name.value();
+        }
+
+        if (ip.has_value())
+        {
+            bodyJson["ip"] = ip.value();
+        }
+
+        http::response<http::dynamic_body> res =
+            sendRequest(managerIp, port, http::verb::post, "/agents", token, bodyJson.dump(), "");
+        return res.result_int();
+    }
+
+} // namespace registration
+
+bool RegisterAgent(const std::string& user,
+                   const std::string& password,
+                   const std::optional<std::string>& name,
+                   const std::optional<std::string>& ip)
+{
+    const auto uuid = "01910343-4ab2-7d94-ab6f-ff76dbb1677d";
+
+    const auto configurationParser = std::make_shared<configuration::ConfigurationParser>();
+    const auto managerIp = configurationParser->GetConfig<std::string>("agent", "manager_ip");
+    const auto port = configurationParser->GetConfig<std::string>("agent", "port");
+
+    auto [authResultCode, token] = registration::SendAuthenticationRequest(managerIp, port, user + ":" + password);
+    if (authResultCode != 200)
```

Comment:
We can use https://www.boost.org/doc/libs/1_85_0/boost/beast/http/status.hpp error codes instead of literals.

Reply:
Done!